ynr/apps/sopn_parsing/helpers/extract_pages.py

```diff
@@ -18,6 +21,88 @@ def extract_pages_for_ballot(ballot, manual_upload=False):
     :type ballot: candidates.models.Ballot
 
     """
+    if ballot.sopn.uploaded_file.path.endswith(".pdf"):
+        pass
+    elif ballot.sopn.uploaded_file.path.endswith(".html"):
+        print(
+            f"Warning! HTML file types are not supported; trying to convert {ballot.sopn.uploaded_file.path} to pdf"
```

Comment:
I'm not sure how successful pandoc is at converting the non-PDF files to types that are then usable by us, and if it works well thats great, but I also think adding validation to the SOPN upload view so that an error message is shown to them if they try to upload a non-PDF is worth considering.

Reply:
I think it may be more successful for docx files than html files, but I see what you're saying. Just to be clear, are you suggesting adding validation in the view just for files that aren't covered here (pdf, docx, html)?

Reply:
1. "less common" way is interesting. The website manual upload is the only way we get by-election SOPNs, so in that sense it's used every week, vs once a year for the sheet importer. In terms of _numbers_ the sheets importer is the thing that gets most SOPNs into the system

2. The person adding to the Google sheet doesn't get feedback directly, but the importer is only run manually by a DC dev on SOPN day. They get to see the output, so I think it's useful to log something. I also think it's useful to add a `--convert-non-pdf`flag to the importer. The default would be not to convert, allowing the dev running the script to check with Slack / the source website to see if there are PDFs. If there aren't, then we could run with the flag. 

Let's have a chat about that change, as we might want to add more flags to control imports as part of that work